### DISCUSSION TITLE
Should one prefer h3:// (DoH) or tls:// (DoT) protocol?

### DISCUSSION THREAD

**Discussion by timkgh** [AdguardTeam/AdGuardHome#6976] [Q&A]
Given an upstream supports both, should one prefer `h3://` (DoH) or `tls://` (DoT) protocol?

Any known issues around connection reuse/keep-alive, etc?

**Comment by bcookatpcsd** [comment]
tl:dr; quic and h3 if you can.. 

h3 and quic are udp.. 
tls and https are tcp.. 

udp would be faster but everyone doesn't support.. 

but sometimes knot-resolver and cloudflare is always faster :) 

![image](https://github.com/user-attachments/assets/96a2332d-3553-42ee-98be-0ab50e4ca5fb)

and as always ymmv.. 

icmp isn't dns.. 

![image](https://github.com/user-attachments/assets/b5076ba7-c186-4d59-a6b1-53295bb79bd6)

![image](https://github.com/user-attachments/assets/fc399c55-3270-4b30-9488-d3218fd77d11)

HTH
